sklearn/impute/tests/test_impute.py

```diff
@@ -1524,6 +1524,20 @@ def test_iterative_imputer_keep_empty_features(initial_strategy):
     assert_allclose(X_imputed[:, 1], 0)
 
 
+def test_iterative_imputer_constant_fill_value():
+    # Test initial imputation using the constant strategy.
+    X = np.array([[-1, 2, 3, -1], [4, -1, 5, -1], [6, 7, -1, -1], [8, 9, 0, -1]])
+
+    X_true = np.array([[0, 2, 3, 0], [4, 0, 5, 0], [6, 7, 0, 0], [8, 9, 0, 0]])
+
+    imputer = IterativeImputer(
+        missing_values=-1, initial_strategy="constant", fill_value=0, max_iter=0
+    )
+    X_trans = imputer.fit_transform(X)
+
+    assert_array_equal(X_trans, X_true)
```

Comment:
Let's focus the test only on checking the value of the fill value. We will let the other test check for the consistency of the output of the transformer.

```suggestion
    X = np.array([[-1, 2, 3, -1], [4, -1, 5, -1], [6, 7, -1, -1], [8, 9, 0, -1]])

    fill_value = 100
    imputer = IterativeImputer(
        missing_values=-1, initial_strategy="constant", fill_value=100, max_iter=0
    )
    imputer.fit_transform(X)
    assert_array_equal(imputer.initial_imputer_.statistics_, fill_value)
```